cpp/tests/test_state_age_function.cpp

```diff
@@ -24,241 +24,382 @@
+    EXPECT_EQ(wrapper.get_scale(), 1.5);
+    EXPECT_EQ(wrapper.get_support_max(dt), 50);
+    EXPECT_NEAR(wrapper.get_mean(), 7.25, 1e-4);
 }
 
 TEST(TestStateAgeFunction, testComparisonOperator)
 {
     // Check that StateAgeFunctions are only considered equal if they are of the same derived
     // class and have the same parameter.
-    mio::ExponentialDecay expdecay(0.5);
-    mio::ExponentialDecay expdecay2(0.5);
-    mio::ExponentialDecay expdecay3(1.0);
+    mio::GammaSurvivalFunction gamma(0.5, 0, 1);
+    mio::GammaSurvivalFunction gamma2(0.5, 0, 1);
+    mio::GammaSurvivalFunction gamma3(2, 0, 1);
+    mio::GammaSurvivalFunction gamma4(0.5, 1.5, 1);
+    mio::GammaSurvivalFunction gamma5(0.5, 0, 2);
     mio::SmootherCosine smoothcos(0.5);
 
-    EXPECT_TRUE(expdecay == expdecay2);
-    EXPECT_FALSE(expdecay == expdecay3);
-    EXPECT_FALSE(expdecay == smoothcos);
+    EXPECT_TRUE(gamma == gamma2);
+    EXPECT_FALSE(gamma == gamma3);
+    EXPECT_FALSE(gamma == gamma4);
+    EXPECT_FALSE(gamma == gamma5);
+    EXPECT_FALSE(gamma == smoothcos);
 
-    // Check that it also holds when a StateAgeFunctionWrapper is set with the respective functions.
-    mio::StateAgeFunctionWrapper wrapper(expdecay);
-    mio::StateAgeFunctionWrapper wrapper2(expdecay2);
-    mio::StateAgeFunctionWrapper wrapper3(expdecay3);
-    mio::StateAgeFunctionWrapper wrapper4(smoothcos);
+    // Check that it also holds when a StateAgeFunctionWrapper is set with the respective functions
+    mio::StateAgeFunctionWrapper wrapper(gamma);
+    mio::StateAgeFunctionWrapper wrapper2(gamma2);
+    mio::StateAgeFunctionWrapper wrapper3(gamma3);
+    mio::StateAgeFunctionWrapper wrapper4(gamma3);
+    mio::StateAgeFunctionWrapper wrapper5(gamma3);
+    mio::StateAgeFunctionWrapper wrapper6(smoothcos);
 
     EXPECT_TRUE(wrapper == wrapper2);
     EXPECT_FALSE(wrapper == wrapper3);
     EXPECT_FALSE(wrapper == wrapper4);
-}
\ No newline at end of file
+    EXPECT_FALSE(wrapper == wrapper5);
+    EXPECT_FALSE(wrapper == wrapper6);
+}
+
+TEST(TestStateAgeFunction, testGamma)
+{
+    // Check if connection between Gammasurvivalfunction and ErlangDensity is fulfilled.
+    ScalarType rate[]  = {0.5, 0.8, 1, 3};
+    int shape[]        = {1, 3, 5, 10, 20};
+    ScalarType times[] = {0, 0.5, 3, 5.555, 20, 70.34};
+    for (int r = 0; r < 4; r++) {
+        for (int s = 0; s < 5; s++) {
+            mio::GammaSurvivalFunction survival(shape[s], 0, 1. / (rate[r]));
+            EXPECT_EQ(survival.eval(0), 1.0);
+            mio::ErlangDensity density(1, 1. / (rate[r]));
+            EXPECT_EQ(density.eval(-1), 0.0);
+
+            for (int tau = 0; tau < 6; tau++) {
+                ScalarType f = 0;
+                for (int k = 1; k < shape[s] + 1; k++) {
+                    density.set_parameter(k);
+                    f += density.eval(times[tau]);
+                }
+                EXPECT_NEAR(f * (1 / rate[r]), survival.eval(times[tau]), 1e-10);
```

Comment:
What property are you checking here exactly?

Reply:
There is a formula that connects the Survivalfunction of an Erlang-distribution with the density. I will add a comment , but it is too much to state the whole formula.